lib/importer/factory/object_factory.rb

```diff
@@ -25,11 +25,16 @@ def run
         object
       end
 
+      ## FOR CONSIDERATION: handle a row (i.e. Work) with more than one file:
+      ## currently the file_set is replaced on update
       def update
         raise "Object doesn't exist" unless object
+        @attr = update_attributes
         run_callbacks(:save) do
-          work_actor.update(environment(update_attributes))
+          work_actor.update(environment(@attr))
         end
+        destroy_existing_file_set if object.file_sets.present?
```

Comment:
Are you sure you want to destroy an existing file set?  Why not update it?  If you replace one fileset with another, people who have cited an existing URI are going to have their references broken.  What's the benefit?

Reply:
Good question; I am not sure that is the best way forward. This is how I envision the use of the csv importer: 
- If I want to update a work with an added file, I would add columns to my csv. I could keep the 1st file or reorder file columns to pick a different one (the thumbnail will be rendered from the 1st).
- If I delete a file from the csv I would expect not to have it appear in the work anymore.

Reply:
I'm not sure that's universal though.  What if I only want to update titles. Reuploading all the files is going to be very slow.  And I don't want to have to do that just to update some metadata.

Reply:
Hi! Apologies for a late reply. Good point.
The method actually destroys the file set only if the file itself needs to be changed - the check is line 142 `f.destroy if attributes[:file] != f.title`. I will rename this method for clarity.
But the file _is_ re-uploaded; which is inefficient for sure. I'll make changes.